ext/odbc/php_odbc.c

```diff
@@ -2046,35 +2084,39 @@ PHP_FUNCTION(odbc_pconnect)
 /* }}} */
 
 /* {{{ odbc_sqlconnect */
-int odbc_sqlconnect(odbc_connection **conn, char *db, char *uid, char *pwd, int cur_opt, int persistent)
+bool odbc_sqlconnect(zval *zv, char *db, char *uid, char *pwd, int cur_opt, int persistent, zend_string *hash)
 {
 	RETCODE rc;
+	odbc_link *link;
+
+	object_init_ex(zv, odbc_connection_ce);
+	link = Z_ODBC_LINK_P(zv);
+	link->connection = (odbc_connection *)pemalloc(sizeof(odbc_connection), persistent);
+	memset(link->connection, 0, sizeof(odbc_connection));
```

Comment:
```suggestion
	link->connection = (odbc_connection *)pecalloc(sizeof(odbc_connection), persistent);
```
Surely we have a calloc version that does those together?